testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -598,6 +598,13 @@ void testCastBooleanToNumeric(CastType castType, SqlOperatorFixture f) {
         "Cast function cannot convert value of type BOOLEAN to type DECIMAL\\(19, 0\\)", false);
   }
 
+  /** Test case for <a href="https://issues.apache.org/jira/projects/CALCITE/issues/CALCITE-6322">
+   *  [CALCITE-6322] Casts to DECIMAL types are ignored</a>. */
+  @Test void testDecimalDecimalCast() {
+    final SqlOperatorFixture f = fixture();
+    f.checkScalar("CAST(1.123 AS DECIMAL(4, 0))", "1", "DECIMAL(4, 0) NOT NULL");
+  }
+
```

Comment:
Instead of adding a new method should we rather enrich `testCastToExactNumeric`?